CRT.c

```diff
@@ -22,6 +22,9 @@ in the source distribution for its full text.
 #include "ProvideCurses.h"
 #include "XUtils.h"
 
+#include <ncurses.h>
+#include <term.h>
```

Comment:
Same goes for `term.h`: Please extend `ProvideCurses.h` accordingly instead. There should be defines available from `configure.ac` …

Reply:
I don't think `configure.ac` currently has defines for `term.h`. Also, I couldn't find any configuration options in the [documentation](https://www.gnu.org/software/autoconf-archive/The-Macros.html) that are to be used for `term.h`. I could only find options for `curses.h`, and its extensions.

https://www.gnu.org/software/autoconf-archive/ax_with_curses.html
https://www.gnu.org/software/autoconf-archive/ax_with_curses_extra.html

Reply:
There actually should be a check for `term.h` be added to `configure.ac` (near where it looks for `ncurses`/`ncursesw`. Also `term.h` should then be added through `ProvideCurses.h`. In the folder `iwyu` the ncurses rules need to be updated accordingly to declare `ProvideCurses.h` is providing the respective symbols from `term.h` instead.

Reply:
Okay. Sounds good.

Reply:
For example, when `term.h` checks are kept in `ProvideCurses.h`, the following compilation error gets thrown:

```
In file included from ProvideCurses.h:33,
                 from CRT.h:15,
                 from CRT.c:12:
HeaderLayout.h:35:12: error: expected ‘)’ before ‘*’ token
   35 |    uint8_t columns;
      |            ^~~~~~~
```

This is because `columns` is a defined variable in the `term.h` header file but refers to another variable in `HeaderLayout.h`

I thought of using `#ifdef` guards to make `term.h` get included only in the `CRT.c` file. But that gives issues as well. Suppose, I use a macro `#ifdef INCLUDE_TERM` and use that in `ProvideCurses.h` to include `term.h` only when required.

In `CRT.h`, the header files are included as shown below:
```
1    #include "config.h" // IWYU pragma: keep
2    
3    #include "CRT.h"
4    
5    #include <errno.h>
6    #include <fcntl.h>
7    #include <langinfo.h>
8    #include <signal.h>
9    #include <stdarg.h>
10   #include <stdio.h>
11   #include <stdlib.h>
12   #include <string.h>
13   #include <unistd.h>
14    
15   #include "ProvideCurses.h"
16   #include "XUtils.h"
```

If I insert `#define INCLUDE_TERM` on line 14, it won't have any effect because `ProvideCurses.h` is included implicitly through `CRT.h`. If I insert the include statement before line 3, it'll will clash with other header files that are implicitly included through `CRT.h`.

Reply:
As said in my previous note, splitting this into `ProvideCurses.h` and `ProvideTerm.h` is fine if that makes things easier.